### PR TITLE
Improvements to the Polaris CLI

### PR DESCRIPTION
**Pull request by eric-maynard**
# Description

This PR covers various improvements to the Polaris CLI. Changes here include:

* Added the `namespaces` command for creating, listing, and dropping namespaces with the CLI
* Refactored error handling to reduce the proliferation of string literals
* Added support for the `FILE` storage type
* Added end-to-end regression tests for the CLI
* Various usability and bug fixes
* Support for the `CLIENT_ID` and `CLIENT_SECRET` environment variables
* CLI documentation

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Added new test `test_cli.py` and updated existing CLI test

# Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/command-line-interface.md [hunk 1]
```diff
@@ -0,0 +1,1085 @@
...
+    Positional arguments:
+      catalog
+```
+
+##### Examples
+
+```
+polaris catalogs delete some_catalog
+
+polaris catalogs delete another_catalog
```
Comment (sfc-gh-aixu) [inline docs/command-line-interface.md:164]: nit: Why do we need `polaris catalogs delete another_catalog` as an example? Seems the same as` polaris catalogs delete some_catalog`.

Comment (sfc-gh-aixu) [reply docs/command-line-interface.md:164]: Oh. seems we are trying to show two examples?

File: docs/command-line-interface.md [hunk 2]
```diff
@@ -0,0 +1,1085 @@
...
+#### update
+
+The `update` subcommand is used to update a catalog. Currently, this command supports changing the properties of a catalog or updating its storage configuration.
+
+```
+input: polaris catalogs update --help
+options:
+  update
+    Named arguments:
+      --default-base-location  (Required) Default base location of the catalog
```
Comment (sfc-gh-aixu) [inline docs/command-line-interface.md:216]: It's said `required`, but the following example doesn't have that. This is not required, right?

`polaris catalogs update --property tag=new_value my_catalog`

File: regtests/t_cli/src/test_cli.py [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+    try:
+        f()
+        throws = False
+    except Exception as e:
+        assert exception_str in str(e)
+    assert throws
+
+
+def get_token(client_id, client_secret):
+    url = 'http://polaris:8181/api/catalog/v1/oauth/tokens'
```
Comment (sfc-gh-aixu) [inline regtests/t_cli/src/test_cli.py:77]: nit: maybe define as a constant.

**Review by sfc-gh-aixu — APPROVED** [review]
LGTM.

**Review by sfc-gh-aixu — APPROVED** [review]

**Review by dennishuo — APPROVED** [review]

**Pull request merged**